src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -86,115 +84,74 @@ export function DiscoverMainRoute(props: Props) {
     id: id || 'new',
   });
 
-  const loadDefaultOrCurrentDataView = useCallback(
-    async (nextSavedSearch: SavedSearch) => {
-      try {
-        const hasUserDataViewValue = await data.dataViews.hasData
-          .hasUserDataView()
-          .catch(() => false);
-        const hasESDataValue =
-          isDev || (await data.dataViews.hasData.hasESData().catch(() => false));
-        setHasUserDataView(hasUserDataViewValue);
-        setHasESData(hasESDataValue);
-
-        if (!hasUserDataViewValue) {
-          setShowNoDataPage(true);
-          return;
-        }
-
-        let defaultDataView: DataView | null = null;
-        try {
-          defaultDataView = await data.dataViews.getDefaultDataView({ displayErrors: false });
-        } catch (e) {
-          //
-        }
-
-        if (!defaultDataView) {
-          setShowNoDataPage(true);
-          return;
-        }
-
-        const { appState } = getDiscoverStateContainer({
-          history,
-          savedSearch: nextSavedSearch,
-          services,
-        });
-        const { index, query } = appState.getState();
-        const ip = await loadDataView(
-          data.dataViews,
-          config,
-          index,
-          historyLocationState?.dataViewSpec
-        );
-
-        const ipList = ip.list;
-        const isTextBasedQuery = query && isOfAggregateQueryType(query);
-        const dataViewData = resolveDataView(
-          ip,
-          nextSavedSearch.searchSource,
-          toastNotifications,
-          isTextBasedQuery
-        );
-        setDataViewList(ipList);
+  const checkData = useCallback(async () => {
```

Comment:
Now when pressing "New", current data view gets replaced by the first data view. It was not the case in 8.7
![changed-data-view](https://user-images.githubusercontent.com/1415710/232762760-6831145d-88b8-415e-ba2f-0b87db4af48a.gif)

Reply:
Thx, i've fixed it. We should cover this case with unit tests however this needs a bit of refactoring, that I'd suggest to do in a follow up

Reply:
Fix: https://github.com/elastic/kibana/pull/153528/commits/c122a2a6fadf2b5800067b2995199f3b2642a0df

Reply:
If I'm understanding correctly, it seems this is still happening or has come back again:
![new_search](https://user-images.githubusercontent.com/25592674/233819537-7f26aaab-57ce-4b82-b445-bb215ddecb07.gif)

Reply:
Thx for catching, that's another case, switching from a persisted saved search to New ... fixed and I've improved a functional to catch this: https://github.com/elastic/kibana/pull/153528/commits/13a79bf4a00acaaa4b7dcdeef1ca37ac4dc6b919